src/cow-react/modules/limitOrders/hooks/useRateImpact.ts

```diff
@@ -0,0 +1,15 @@
+import { useAtomValue } from 'jotai/utils'
+import { limitRateAtom } from '@cow/modules/limitOrders/state/limitRateAtom'
+import { useMemo } from 'react'
+
+export function useRateImpact(): number {
+  const { activeRate, executionRate, isLoading, isLoadingExecutionRate } = useAtomValue(limitRateAtom)
+
+  return useMemo(() => {
+    if (!activeRate || !executionRate || isLoading || isLoadingExecutionRate) return 0
+
+    const ratePercent = +activeRate.divide(executionRate).multiply(100).subtract(100).toFixed(1)
```

Comment:
In that case, `rate percent` it's a number in the range: `-100` - `+999`, so `Number` type is enough here.
Since it's a percent we don't need to much decimals, so I trim it using `toFixed(1)`.